NewsFeed/views/CCBRNewsViewController.m

```diff
@@ -83,9 +93,41 @@ - (void)viewDidLoad {
 }
 
 - (void)updateUI {
+    self.loadingIndicatorView.hidden = YES;
     self.collectionView.hidden = self.viewModel.collectionViewHidden;
     self.errorMessageLabel.hidden = self.viewModel.errorMessageLabelHidden;
     [self.collectionView reloadData];
+    [self.collectionView.infiniteScrollingView stopAnimating];
+}
+
+- (void)loadDataError: (NSError*)error {
+    NSString* message = @"API errors";
+    if (![error.localizedDescription isEqualToString:@""]) {
+        message = error.localizedDescription;
+    }
+    if (self.viewModel.collectionViewHidden && !self.viewModel.errorMessageLabelHidden) {
+        self.errorMessageLabel.text = message;
+    } else {
+        [self showMessage:message];
+    }
+    self.loadingIndicatorView.hidden = YES;
+    self.collectionView.hidden = self.viewModel.collectionViewHidden;
+    self.errorMessageLabel.hidden = self.viewModel.errorMessageLabelHidden;
```

Comment:
These 2 lines unnecessarily duplicate the logic of `updateUI` method. You should unify the 2 methods somehow.

Reply:
Hi Mr. Tung,
I have just updated my code from your feedback. The operating flow would be:
- If collectionView no data, error lable will show error message.
- If load more error, i will use alert to show error message.
- Loading only show at start application.
Please review my changes.
Thank you!